File: klasa 2/python/silnia.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potega.py
#
# Obliczanie silni liczby naturalnej.


def silnia_it(n):
    """
    Funkcjca oblicza silnięliczby naturalnej
    n! = n*(n-1)
    """
    wynik = 1

    for i in range(1,n+1):
        wynik = n * (n-1)
        #print(wynik)
        
    return wynik   
    
    pass
def main(args):
    
    #a = int(input("Podaj podstawę potęgi")
    #n = int(input("Podaj wykładnik potęgi")
    n =  int(input("Podaj podstawę silni: "))
    print("{}! wynosi {} ".
           format(n,silnia_it(n)))
    
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: klasa 2/c++/silnia.cpp
/*
 * silnia.cpp
 */


#include <iostream>
using namespace std;


int silnia_it(int a) {
    
    int wynik = 1;
    for (int i = 1; i < a+1; i++) {
        wynik = a * (a-1);
        }
    return wynik;
    }

int silnia_rek(int a){
    
    if (a == 1)
        return 1;
    return a * silnia_rek(a-1);
    }

int main(int argc, char **argv)
{
	int a;
    cout << "Podaj liczbę: " << endl;
    cin >> a;
    cout << "Silnia dla: " << a << " wynosi: " << silnia_it(a) << endl;
    cout << "Silnia dla: " << a << " wynosi: " << silnia_rek(a) << endl;
	return 0;
}

